numba/cuda/cudadecl.py

```diff
@@ -105,6 +105,69 @@ class Cuda_threadfence_system(ConcreteTemplate):
     cases = [signature(types.none)]
 
 
+@intrinsic
+class Cuda_popc(ConcreteTemplate):
+    """
+    Supported types from http://docs.nvidia.com/cuda/nvvm-ir-spec/index.html#bit-manipulations-intrinics
+    """
+    key = cuda.popc
+    cases = [
+        signature(types.int8, types.int8),
+        signature(types.int16, types.int16),
+        signature(types.int32, types.int32),
+        signature(types.int64, types.int64),
+        signature(types.uint8, types.uint8),
+        signature(types.uint16, types.uint16),
+        signature(types.uint32, types.uint32),
+        signature(types.uint64, types.uint64),
+    ]
+
+
+@intrinsic
+class Cuda_brev(ConcreteTemplate):
+    key = cuda.brev
+    cases = [
+        signature(types.uint32, types.uint32),
+        signature(types.uint64, types.uint64),
+    ]
+
+
+@intrinsic
+class Cuda_clz(ConcreteTemplate):
+    """
+    Supported types from http://docs.nvidia.com/cuda/nvvm-ir-spec/index.html#bit-manipulations-intrinics
+    """
+    key = cuda.clz
+    cases = [
+        signature(types.int8, types.int8),
+        signature(types.int16, types.int16),
+        signature(types.int32, types.int32),
+        signature(types.int64, types.int64),
+        signature(types.uint8, types.uint8),
+        signature(types.uint16, types.uint16),
+        signature(types.uint32, types.uint32),
+        signature(types.uint64, types.uint64),
+    ]
+
+
+@intrinsic
+class Cuda_ffs(ConcreteTemplate):
+    """
+    Supported types from http://docs.nvidia.com/cuda/nvvm-ir-spec/index.html#bit-manipulations-intrinics
```

Comment:
This link doesn't appear to have a reference to `ffs`?

Also, the docstring https://github.com/numba/numba/pull/2824/files#diff-1c116a52e2457a8cebe1e9d2072d195dR320 suggests 32bit only, but there's 64bit types below?

Reply:
I've made the docs a bit clearer - nb. the CUDA intrinsics are named after the PTX intrinsic (so `ffs`), not the LLVM ones (`cttz` here). Where there's not a 1:1 mapping (like `brev` currently) we have to use the CUDA name, and using a mixture of CUDA & LLVM variables for the intrinsics wouldn't be very user-friendly!